pkg/asset/cluster/metadata.go

```diff
@@ -0,0 +1,107 @@
+package cluster
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/openshift/installer/pkg/asset"
+	"github.com/openshift/installer/pkg/asset/cluster/aws"
+	"github.com/openshift/installer/pkg/asset/cluster/libvirt"
+	"github.com/openshift/installer/pkg/asset/cluster/openstack"
+	"github.com/openshift/installer/pkg/asset/installconfig"
+	"github.com/openshift/installer/pkg/types"
+	"github.com/pkg/errors"
+)
+
+const (
+	metadataFileName = "metadata.json"
+)
+
+// Metadata contains information needed to destroy clusters.
+type Metadata struct {
+	file *asset.File
+}
+
+var _ asset.WritableAsset = (*Metadata)(nil)
+
+// Name returns the human-friendly name of the asset.
+func (m *Metadata) Name() string {
+	return "Metadata"
+}
+
+// Dependencies returns the direct dependencies for the metadata
+// asset.
+func (m *Metadata) Dependencies() []asset.Asset {
+	return []asset.Asset{
+		&installconfig.ClusterID{},
+		&installconfig.InstallConfig{},
+	}
+}
+
+// Generate generates the metadata asset.
+func (m *Metadata) Generate(parents asset.Parents) (err error) {
+	clusterID := &installconfig.ClusterID{}
+	installConfig := &installconfig.InstallConfig{}
+	parents.Get(clusterID, installConfig)
+
+	if installConfig.Config.Platform.None != nil {
+		return errors.New("metadata cannot be created with platform set to 'none'")
```

Comment:
No error should be returned here. Otherwise, the `ignition-configs` target will fail for a `None` platform.

Reply:
> No error should be returned here.

Fixed with  a9afdc9 -> e523949.